src/Info.php

```diff
@@ -37,108 +33,45 @@
     /** @var string */
     private $version;
 
-    public function __construct($data = [])
-    {
-        $this->contact = new Contact();
-        $this->license = new License();
-
-        $this->merge($data);
-    }
-
-    protected function doMerge($data, $overwrite = false)
+    public function __construct(array $data)
     {
-        MergeHelper::mergeFields($this->title, $data['title'] ?? null, $overwrite);
-        MergeHelper::mergeFields($this->termsOfService, $data['termsOfService'] ?? null, $overwrite);
-        MergeHelper::mergeFields($this->version, $data['version'] ?? null, $overwrite);
-
-        $this->contact->merge($data['contact'] ?? [], $overwrite);
-        $this->license->merge($data['license'] ?? [], $overwrite);
-
-        $this->mergeDescription($data, $overwrite);
-        $this->mergeExtensions($data, $overwrite);
+        $this->title = $data['title'];
```

Comment:
We shouldn't fail here imo, the title may be set later

Reply:
Well yeah there are a lot of objects and properties that you would want to populate later, in that case we will have to validate everything afterwards which is bad idea I think. If you want to create """empty""" Info object and let end dev to set the values you can either set it as an empty string (which is valid), or and he can use setter any time he want (it's not there yet, but ofc will be)

Reply:
Well the goal of the library is to build an OpenApi documentation, so for me the document may be invalid while built, the only thing that matters is that it's valid when `toArray` is called (in the current version there is no validation, I think we can do the same at first here too).